### PR TITLE
Svr file structure

### PR DESCRIPTION
**Pull request by AmGthrow**
Rebuilt the server using the file structure from #49, the Google Spreadsheet now uses .env variables instead of JSON and functionality is more decomposed 

- API response logic is handled by `\server\controller`
- Database interactions are handled by `\server\database`
- API endpoints and documentation are in `\server\route`

BREAKING CHANGE:
- Spreadsheet interaction in `\server\database\spreadshet.js` now uses environment variables. This means we're using a new `.env` file

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -1,77 +1,42 @@
...
-
-app.use(
-    express.urlencoded({
-        extended: true
-    })
-)
-
 // Serve documentation for API endpoints on /api-docs
-app.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs))
+app.use('/api/docs', swaggerUi.serve, swaggerUi.setup(swaggerDocs))
```
Comment (AmGthrow) [inline server/server.js:36]: Documentation middleware, should this stay in `server.js` or somewhere in `\server\middleware`?

Comment (Carryl21) [reply server/server.js:36]: we can use the middleware folder for that

Comment (Carryl21) [reply server/server.js:36]: just to make everything clear and not puno

Comment (AmGthrow) [reply server/server.js:36]: You mind sharing the link you got the idea for the file organization from? I'm not sure how to move this over to `\middleware` kasi, I can't just `app.use()` when `app` is in a diff module

Comment (Carryl21) [reply server/server.js:36]: as for the link, i forgot asa to nako nakita. i just found this trick last year man gud hihi

Comment (Carryl21) [reply server/server.js:36]: here's a sample code for that. however, i'm not sure if this will work. 

image

Comment (AmGthrow) [reply server/server.js:36]: Got it, thanks! Thought middleware would also have to include mounting

**Pull request merged**